test/metabase/test/data/dataset_definitions.clj

```diff
@@ -208,6 +208,27 @@
        (t/offset-time t)                ; time-tz
        cnt])]])                         ; num-crows
 
+(tx/defdataset dt-attempted-murders
```

Comment:
Instead of creating more datasets which slows down tests a lot since loading them is really expensive I think we should just make Oracle work with the existing `attempted-murders` dataset, I think it's ok to have Oracle create `time` as whatever column type (maybe a datetime with 1970-01-01 for the date part, or maybe `TEXT` with all `NULL` values, doesn't really matter) and still skip those tests anyway. For the new test you're adding here it shouldn't matter since we're not looking at values of that column anyway

At some point I think it would be nice to combine all of the datasets into one mega dataset and then databases can just load what they can from it

Reply:
honestly for this bug tho I think it would have been fine to write a Oracle-only test in `metabase.driver.oracle-test` that runs a native query like 

```sql
SELECT TIMESTAMP WITH LOCAL TIME ZONE '<whatever>' AS ts
FROM dual
```

and makes sure it's returned correctly. Creating a new dataset is pretty heavy handed

Reply:
> honestly for this bug tho I think it would have been fine to write a Oracle-only test

I could write an oracle-only test. But because I noticed there's no test to enforce [this behaviour](https://github.com/metabase/metabase/pull/44456/files#r1648008957) across all drivers that support `:type/DateTimeWithLocalTZ` types I wanted to try my hand at that.

I'd prefer to do the approach you described above with Oracle creating `NULL` values for the base types it doesn't support.

Reply:
yeah, writing both tests is even better. I wrote an Oracle test in one of the comments above so you can use that if you want

Reply:
I've removed this dataset now. I'm only testing oracle with select statements as you suggested.